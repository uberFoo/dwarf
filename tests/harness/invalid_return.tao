// Main is valid, just to not confuse the issue.
fn main() {}

// This function indicates a value of int, but it's implicitly empty.
fn a() -> int {}

// This function is returning a value of the wrong type.
fn b() -> int {
    "hello"
}

// This function could coerce, but it should not.
fn c() -> string {
    42
}
