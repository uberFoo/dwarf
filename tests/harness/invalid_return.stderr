[31mError:[0m Type mismatch: expected `[3;35mint[0m`, found `[3;35m()[0m`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m invalid_return:5:15 [38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m5 â”‚[0m [38;5;249mf[0m[38;5;249mn[0m[38;5;249m [0m[38;5;249ma[0m[38;5;249m([0m[38;5;249m)[0m[38;5;249m [0m[38;5;249m-[0m[38;5;249m>[0m[38;5;249m [0m[33mi[0m[33mn[0m[33mt[0m[33m [0m[31m{[0m[31m}[0m
 [38;5;240m  â”‚[0m           [33mâ”€[0m[33mâ”€[0m[33mâ”¬[0m[33mâ”€[0m[31mâ”€[0m[31mâ”¬[0m  
 [38;5;240m  â”‚[0m             [33mâ•°[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m expected [36m[3;35mint[0m[0m
 [38;5;240m  â”‚[0m                [31mâ”‚[0m  
 [38;5;240m  â”‚[0m                [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m found [36m[3;35m()[0m[0m
[38;5;246mâ”€â”€â”€â•¯[0m

[31mError:[0m Type mismatch: expected `[3;35mint[0m`, found `[3;35mstring[0m`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m invalid_return:9:5 [38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m8 â”‚[0m [38;5;249mf[0m[38;5;249mn[0m[38;5;249m [0m[38;5;249mb[0m[38;5;249m([0m[38;5;249m)[0m[38;5;249m [0m[38;5;249m-[0m[38;5;249m>[0m[38;5;249m [0m[33mi[0m[33mn[0m[33mt[0m[33m [0m[33m{[0m
 [38;5;240m  â”‚[0m           [33mâ”€[0m[33mâ”€[0m[33mâ”¬[0m[33mâ”€[0m[33mâ”€[0m  
 [38;5;240m  â”‚[0m             [33mâ•°[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m expected [36m[3;35mint[0m[0m
 [38;5;246m9 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[31m"[0m[31mh[0m[31me[0m[31ml[0m[31ml[0m[31mo[0m[31m"[0m
 [38;5;240m  â”‚[0m     [31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m  
 [38;5;240m  â”‚[0m        [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m found [36m[3;35mstring[0m[0m
[38;5;246mâ”€â”€â”€â•¯[0m

[31mError:[0m Type mismatch: expected `[3;35mstring[0m`, found `[3;35mint[0m`.
    [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m invalid_return:14:5 [38;5;246m][0m
    [38;5;246mâ”‚[0m
 [38;5;246m13 â”‚[0m [38;5;249mf[0m[38;5;249mn[0m[38;5;249m [0m[38;5;249mc[0m[38;5;249m([0m[38;5;249m)[0m[38;5;249m [0m[38;5;249m-[0m[38;5;249m>[0m[38;5;249m [0m[33ms[0m[33mt[0m[33mr[0m[33mi[0m[33mn[0m[33mg[0m[33m [0m[33m{[0m
 [38;5;240m   â”‚[0m           [33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”¬[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m  
 [38;5;240m   â”‚[0m               [33mâ•°[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m expected [36m[3;35mstring[0m[0m
 [38;5;246m14 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[31m4[0m[31m2[0m
 [38;5;240m   â”‚[0m     [31mâ”€[0m[31mâ”¬[0m  
 [38;5;240m   â”‚[0m      [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m found [36m[3;35mint[0m[0m
[38;5;246mâ”€â”€â”€â”€â•¯[0m

